### PR TITLE
Shutdown scenario

### PR DESCRIPTION
**Pull request by shhsu**
The scenario works when I manually start the service again after reboot. However, I am still trying to figure out a way to restart the service automatically when the machine reboots. The setup.ps1 doesn't quite seem working currently

### CODE REVIEW

**Comment by shhsu** [comment]
recover_flow
I have redrawn the flow a little.

* Messages would be backed up before command is executed
* If return code exists (and is 0) the flow may be interrupted if the command involve system interruption (e.g. shutdown)
* When the app starts, it looks for recovery files, recover the message, assign exit code 0. The message is then submitted after the recovery file is cleaned up

**Comment by shhsu** [comment]
@SajayAntony @sivagms

**Comment by shhsu** [comment]
Completed

File: README.md [hunk 1]
```diff
@@ -70,6 +74,15 @@ sudo update-rc.d cmdport defaults
 sudo update-rc.d -f cmdport remove
 ```
 
+Optional:
+
+Run `install\linux\register-cron.ps1`. This would register wave to start each time the system starts. Note that this step is required for service shutdown/restart automation scenario.
```
Comment (SajayAntony) [inline README.md]: This is already registered as a part of init script. Why do we need to add this additionally to cron?

**Comment by SajayAntony** [comment]
@martinpf @sivagms could you guys review this . There are some good enhancements to make this a windows service.

File: README.md (hunk 1 above)
Comment (shhsu) [inline README.md]: Since it's in the install script I will remove it.

**Comment by shhsu** [comment]
Actually, since the auto startup logic was already in place, I will just remove them in my PR. This includes the windows service setup

**Comment by shhsu** [comment]
synced up with @SajayAntony, merged

**Pull request merged**